sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java

```diff
@@ -1258,6 +1263,27 @@ public boolean apply(@Nonnull Method input) {
         iface.getName());
   }
 
+  private static boolean checkInheritedFrom(Class<?> checkClass, Class fromClass) {
+    if (checkClass.getInterfaces().length == 0) {
+      return checkClass.equals(fromClass);
+    }
+
+    boolean valid = true;
+
+    for (Class<?> klass: checkClass.getInterfaces()){
+      valid &= checkInheritedFrom(klass, fromClass);
```

Comment:
Please attempt to collect the list of which are leaves in the inheritance trees that don't extend PipelineOptions to improve the quality of the error message so that it contains this list making this error actionable for users.

Reply:
addressed. add a set to collect all interface doesn't inherited from `PipelineOptions`.